src/CMake/VisItMacros.cmake

```diff
@@ -263,3 +263,22 @@ function(ADD_CMAKE_GEN_TARGET gen_name
 endfunction()
 
 
+##############################################################################
+# This macro appends to a CACHE var list denoted by the NAME argument.
+# Caller is responsible for unsetting the CACHE var when no longer needed
+# to avoid polluting the CACHE.
+#
+# Designed mainly for targets whose sources live in subdirectories,
+# as a means for them to add to the parent's list of source/includes/etc
+# to prepare for a 'blt_add_library' call which requires SOURCES.
+#
+##############################################################################
+
+macro(visit_append_list)
+    cmake_parse_arguments(arg "" "NAME" "ITEMS" ${ARGN})
```

Comment:
Is the `ITEMS` label really needed? I mean won't the first thingy after `NAME` be the name and everything else be the items to add to the list? Maybe this way is more cmake-standard?

Reply:
ITEMS isn't needed, neither is NAME, but combined with cmake_parse_arguments, use of named arguments is a way to ensure that required arguments are present. When the macro is called, use of named arguments also makes the functionality of what is being called a bit clearer.